src/common/types/timestamp.cpp

```diff
@@ -61,3 +64,140 @@ void Timestamp::Convert(timestamp_t date, date_t &out_date, dtime_t &out_time) {
 	out_date = GetDate(date);
 	out_time = GetTime(date);
 }
+
+timestamp_t Timestamp::GetCurrentTimestamp() {
+	auto now = std::chrono::system_clock::now();
+	auto in_time_t = std::chrono::system_clock::to_time_t(now);
+
+	std::stringstream ss;
+	ss << std::put_time(std::localtime(&in_time_t), "%Y-%m-%d %X");
```

Comment:
Can't you extract the individual elements (year/month/day/etc)  from the std::chrono object directly? This conversion into time_t and then into a string and then parsing that string again seems inefficient.

Reply:
No, this is the only way that I found to have the formatted date. There is a calendar class but only for C++20. I can get directly the in_time_t, but I still need to convert it to local_time and to get our internal timestamp I can either convert to date and time separately and then timestamp or to string and timestamp afterward.

Reply:
I think you can extract it straight from the tm: https://stackoverflow.com/questions/15957805/extract-year-month-day-etc-from-stdchronotime-point-in-c

You need to use std::localtime, but you don't need to convert it to a string and then parse it back.